src/main/java/org/broadinstitute/hellbender/utils/dataflow/BucketUtils.java

```diff
@@ -74,4 +77,41 @@ public static OutputStream createFile(String path, PipelineOptions popts) {
             throw new UserException.CouldNotCreateOutputFile(path, x);
         }
     }
+
+    /**
+     * Copies a file. Can be used to copy e.g. from GCS to local.
+     *
+     * @param sourcePath the path to read from. If GCS, it must start with "gs://".
+     * @param popts the pipeline's options, with authentication information.
+     * @param destPath the path to copy to. If GCS, it must start with "gs://".
+     * @throws IOException
+     */
+    public static void copyFile(String sourcePath, PipelineOptions popts, String destPath) throws IOException {
+        try (
+            InputStream in = openFile(sourcePath, popts);
+            OutputStream fout = createFile(destPath, popts)) {
+            final byte[] buf = new byte[1024 * 1024];
+            int count;
+            while ((count = in.read(buf)) > 0) {
+                fout.write(buf, 0, count);
+            }
+        }
+    }
+
+    /**
+     * Deletes a file, local or on GCS.
+     *
+     * @param pathToDelete the path to delete. If GCS, it must start with "gs://".
+     * @param popts the pipeline's options, with authentication information.
+     */
+    public static void deleteFile(String pathToDelete, PipelineOptions popts) throws IOException, GeneralSecurityException {
+        if (!BucketUtils.isCloudStorageUrl(pathToDelete)) {
+            boolean ok = new File(pathToDelete).delete();
+            if (!ok) throw new IOException("Unable to delete '"+pathToDelete+"'");
+        }
+        GcsPath path = GcsPath.fromUri(pathToDelete);
+        GcsOptions gcsOptions = (GcsOptions)popts.as(GcsOptions.class);
+        Storage storage = Transport.newStorageClient(gcsOptions).build();
+        storage.objects().delete(path.getBucket(), path.getObject()).execute();
+    }
```

Comment:
At some point we'll have to generalize all of these more so they work with hadoop file system as well.